src/main/java/com/coveros/selenified/services/Check.java

```diff
@@ -161,20 +186,58 @@ JsonObject checkEquals(JsonObject expectedJson) {
      * @param expectedJson - the expected response json array
      */
     JsonArray checkEquals(JsonArray expectedJson) {
-        boolean pass = false;
         JsonArray actualJson = null;
         if (this.response.getArrayData() != null) {
             actualJson = this.response.getArrayData();
-            pass = actualJson.equals(expectedJson);
         }
-        if (pass) {
+        if (expectedJson.equals(actualJson)) {
             this.reporter.pass("", EXPECTED_TO_FIND + DIV_I + Reporter.formatHTML(GSON.toJson(expectedJson)) + END_IDIV, FOUND + Reporter.formatResponse(this.response));
         } else {
             this.reporter.fail("", EXPECTED_TO_FIND + DIV_I + Reporter.formatHTML(GSON.toJson(expectedJson)) + END_IDIV, FOUND + Reporter.formatResponse(this.response));
         }
         return actualJson;
     }
 
+    /**
+     * Checks the actual response json payload contains a key with a value equal to the expected
+     * value. The jsonKeys should be passed in as crumbs of the keys leading to the field with
+     * the expected value. This result will be written out to the output file.
+     *
+     * @param jsonKeys      - the crumbs of json object keys leading to the field with the expected value
+     * @param expectedValue - the expected value
+     */
+    @SuppressWarnings("squid:S1201")
+    abstract void equals(List<String> jsonKeys, Object expectedValue);
+
+    /**
+     * Checks the actual response json payload contains a key with a value equal to the expected
+     * value. The jsonCrumbs should be passed in as crumbs of the keys leading to the field with
+     * the expected value. This result will be written out to the output file.
+     *
+     * @param jsonCrumbs    - the crumbs of json object keys leading to the field with the expected value
+     * @param expectedValue - the expected value
+     */
+    Object checkEquals(List<String> jsonCrumbs, Object expectedValue) {
+        JsonElement actualValue = this.response.getObjectData();
+        for (String jsonCrumb : jsonCrumbs) {
+            if (actualValue == null) {
+                break;
+            }
+            if (!(actualValue instanceof JsonObject)) {
+                actualValue = null;
+                break;
+            }
+            actualValue = actualValue.getAsJsonObject().get(jsonCrumb);
+        }
+        Object objectVal = castObject(expectedValue, actualValue);
+        if (expectedValue.equals(objectVal)) {
+            this.reporter.pass("", EXPECTED_TO_FIND + STARTI + Reporter.formatHTML(String.join(" \uD83E\uDC1A ", jsonCrumbs)) + ENDI + " with value of: " + DIV_I + Reporter.formatHTML(GSON.toJson(expectedValue)) + END_IDIV, FOUND + DIV_I + Reporter.formatHTML(GSON.toJson(objectVal)) + END_IDIV);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal " \uD83E\uDC1A " 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1192)